poly/src/polynomial/univariate/dense.rs

```diff
@@ -378,27 +378,32 @@ impl<'a, F: Field> AddAssign<&'a DensePolynomial<F>> for DensePolynomial<F> {
 
 impl<'a, F: Field> AddAssign<(F, &'a DensePolynomial<F>)> for DensePolynomial<F> {
     fn add_assign(&mut self, (f, other): (F, &'a DensePolynomial<F>)) {
-        if self.is_zero() {
-            self.coeffs.truncate(0);
-            self.coeffs.extend_from_slice(&other.coeffs);
-            self.coeffs.iter_mut().for_each(|c| *c *= &f);
+        // No need to modify self if other is zero
+        if other.is_zero() {
             return;
-        } else if other.is_zero() {
+        }
+
+        // If the first polynomial is zero, just copy the second one and scale by f.
+        if self.is_zero() {
+            self.coeffs = other.coeffs.iter().map(|&c| f * c).collect();
```

Comment:
I prefer the older method because it preserves the allocation of `self`, whereas this approach creates a new allocation.

Actually I realized that the old method doesn't preserve allocations either, and we should be using `self.coeffs.clear()` instead of `self.coeffs.truncate(0)`.

Reply:
yep agreed, just fixed